tensorflow/c/kernels/histogram_summary_op.cc

```diff
@@ -0,0 +1,168 @@
+  ~TensorWrapper() { 
+    TF_DeleteTensor(t);
+  }
+};
+
+struct StatusWrapper { 
+  TF_Status* s; 
+  StatusWrapper() { 
+    s = TF_NewStatus(); 
+  }
+  ~StatusWrapper() { 
+    TF_DeleteStatus(s);
+  }
+};
+typedef struct HistogramSummaryOp {
+  std::string op_node_name; 
+};
+
+static void* HistogramSummaryOp_Create(TF_OpKernelConstruction* ctx) {
+  HistogramSummaryOp* kernel = new HistogramSummaryOp; 
+  TF_StringView string_view_name = TF_OpKernelConstruction_GetName(ctx); 
+  kernel->op_node_name = std::string(string_view_name.data, 
+      string_view_name.len);
+  return kernel; 
+}
+
+static void HistogramSummaryOp_Delete(void* kernel) {
+  delete static_cast<HistogramSummaryOp*>(kernel); 
+}
+
+template<typename T>
+static void HistogramSummaryOp_Compute(void* kernel, TF_OpKernelContext* ctx) {
+  HistogramSummaryOp* k = static_cast<HistogramSummaryOp*>(kernel);
+  TensorWrapper tags_wrapper; 
+  TensorWrapper values_wrapper; 
+  StatusWrapper status_wrapper;
+  TF_GetInput(ctx, 0, &tags_wrapper.t, status_wrapper.s);
+  if (TF_GetCode(status_wrapper.s) != TF_OK) { 
+    TF_OpKernelContext_Failure(ctx, status_wrapper.s);
+    return; 
+  }
+  TF_GetInput(ctx, 1, &values_wrapper.t, status_wrapper.s);
+  if (TF_GetCode(status_wrapper.s) != TF_OK) { 
+    TF_OpKernelContext_Failure(ctx, status_wrapper.s);
+    return; 
+  }
+  if (TF_NumDims(tags_wrapper.t) != 0) { 
+    TF_SetStatus(status_wrapper.s, TF_INVALID_ARGUMENT, "tags must be scalar");
+    TF_OpKernelContext_Failure(ctx, status_wrapper.s);
+    return; 
+  }
+  // Cast values to array to access tensor elements by index 
+  auto values_array = static_cast<T*>(TF_TensorData(values_wrapper.t)); 
+  tensorflow::histogram::Histogram histo; 
+  for (int i = 0; i < TF_TensorElementCount(values_wrapper.t); ++i) { 
+    const double double_val = static_cast<double>(values_array[i]); 
+    if (Eigen::numext::isnan(double_val)) { 
+      std::ostringstream err; 
+      err << "Nan in summary histogram for: " << k->op_node_name; 
+      TF_SetStatus(status_wrapper.s, TF_INVALID_ARGUMENT, err.str().c_str());
+      break;
+    }
+    else if (Eigen::numext::isinf(double_val)) { 
+      std::ostringstream err; 
+      err << "Infinity in Histogram for: " << k->op_node_name; 
+      TF_SetStatus(status_wrapper.s, TF_INVALID_ARGUMENT, err.str().c_str());
+      break; 
+    }
+    histo.Add(double_val);
+  }
+  tensorflow::Summary s; 
+  tensorflow::Summary::Value* v = s.add_value(); 
+  const tensorflow::tstring& tag = *(static_cast<tensorflow::tstring*>(
+      TF_TensorData(tags_wrapper.t))); 
+  v->set_tag(tag.data(), tag.size()); 
+  histo.EncodeToProto(v->mutable_histo(), false /* Drop zero buckets */); 
+
+  // Must use new status for TF_AllocateOutput if previous status is set 
+  // because of an invalid values argument.  
+  StatusWrapper allocation_status_wrapper;
+  TensorWrapper summary_tensor_wrapper; 
+  TF_Tensor* summary_tensor= TF_AllocateOutput(ctx, 0,
```

Comment:
nit. Can merge next line with this one.
`summary_tensor_wrapper.t = TF_AllocateOutput(...`